File: rationwala-api/src/main/java/com/app/rationwala/dto/RegisterRequest.java
package com.app.rationwala.dto;

import com.app.rationwala.model.LoginCredential;
import com.app.rationwala.model.UserProfile;

import lombok.Data;

public @Data class RegisterRequest {

	private LoginCredential loginCredential;
	private UserProfile userProfile;

	public RegisterRequest() {
	}

	public RegisterRequest(LoginCredential loginCredential, UserProfile userProfile) {
		super();
		this.loginCredential = loginCredential;
		this.userProfile = userProfile;
	}

}

File: rationwala-ui/src/app/_services/dto/register-request.ts
import { UserProfile } from "src/app/_models";
import { LoginCredential } from "./login-request";

export class RegisterRequest{
    private userProfile: UserProfile;
    private loginCredential: LoginCredential;
    
    /**
     * Getter $userProfile
     * @return {UserProfile}
     */
	public get $userProfile(): UserProfile {
		return this.userProfile;
	}

    /**
     * Getter $loginCredential
     * @return {LoginCredential}
     */
	public get $loginCredential(): LoginCredential {
		return this.loginCredential;
	}

    /**
     * Setter $userProfile
     * @param {UserProfile} value
     */
	public set $userProfile(value: UserProfile) {
		this.userProfile = value;
	}

    /**
     * Setter $loginCredential
     * @param {LoginCredential} value
     */
	public set $loginCredential(value: LoginCredential) {
		this.loginCredential = value;
	}
    
}